--- src/main/java/me/cortex/vulkanite/lib/cmd/CommandManager.java
+++ src/main/java/me/cortex/vulkanite/lib/cmd/CommandManager.java
@@ -226,7 +226,11 @@

         public void waitForExecutions(int execQueue, List<Long> executions) {
             synchronized (waitingFor) {
-                long execMax = executions.stream().mapToLong(Long::longValue).max().orElse(waitingFor.getOrDefault(execQueue, 0));
+                long execMax = waitingFor.getOrDefault(execQueue, 0);
+                for (int i = 0; i < executions.size(); i++) {
+                    long exec = executions.get(i);
+                    if (exec > execMax) execMax = exec;
+                }
                 waitingFor.put(execQueue, execMax);
             }
         }
